### PR TITLE
CP-3677 - Filter users by role and status

### PR DESCRIPTION
**Pull request by Vikeln**
## Summary by CodeRabbit

* **New Features**
  * Added user filtering by role and by enabled status in user listings.

* **Tests**
  * Added comprehensive tests covering role, enabled, combined filters, and edge cases.

* **Chores**
  * Bumped project version to 1.1.12.
  * Added release note entry noting compatibility with Mifos Version 1.5.0.

### CODE REVIEW

**Review by simiyu17 — APPROVED** [review]

File: src/main/java/org/apache/fineract/api/UsersApi.java [hunk 1]
```diff
@@ -67,9 +68,22 @@ public class UsersApi {
...
+    public List<AppUser> retrieveAll(@RequestParam(required = false) String role,
+                                     @RequestParam(required = false) Boolean enabled) {
         ThreadLocalContextUtil.getCurrentUser().validateHasReadPermission(this.resourceNameForPermissions);
-        return this.appuserRepository.findAll();
+        Specification<AppUser> spec = Specification.where(null);
+        final String normalizedRole = role == null ? null : role.trim();
+        if (normalizedRole != null && !normalizedRole.isEmpty()) {
+            spec = spec.and((root, query, cb) -> {
+                query.distinct(true);
+                return cb.equal(root.join("roles").get("name"), normalizedRole);
```
Comment (amydoxym) [inline src/main/java/org/apache/fineract/api/UsersApi.java:79]: is this being sanitized somewhere to avoid SQL injection? Does Spring Jpa Specification do it in the background?

Comment (Vikeln) [reply src/main/java/org/apache/fineract/api/UsersApi.java:79]: yes @amydoxym

**Review by amydoxym — APPROVED** [review]
